core/optaplanner-core-impl/src/test/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/ListChangeMoveSelectorTest.java

```diff
@@ -52,25 +56,25 @@ void original() {
         assertAllCodesOfMoveSelector(moveSelector,
                 // Moving 3 from C[0]
                 "3 {C[0]->A[0]}",
-                "3 {C[0]->A[1]}",
-                "3 {C[0]->A[2]}",
                 "3 {C[0]->B[0]}",
                 "3 {C[0]->C[0]}", // noop
                 "3 {C[0]->C[1]}", // undoable
+                "3 {C[0]->A[2]}",
+                "3 {C[0]->A[1]}",
                 // Moving 1 from A[1]
                 "1 {A[1]->A[0]}",
-                "1 {A[1]->A[1]}", // noop
-                "1 {A[1]->A[2]}", // undoable
                 "1 {A[1]->B[0]}",
                 "1 {A[1]->C[0]}",
                 "1 {A[1]->C[1]}",
+                "1 {A[1]->A[2]}", // undoable
+                "1 {A[1]->A[1]}", // noop
                 // Moving 2 from A[0]
                 "2 {A[0]->A[0]}", // noop
-                "2 {A[0]->A[1]}",
-                "2 {A[0]->A[2]}", // undoable
                 "2 {A[0]->B[0]}",
                 "2 {A[0]->C[0]}",
-                "2 {A[0]->C[1]}");
+                "2 {A[0]->C[1]}",
+                "2 {A[0]->A[2]}", // undoable
+                "2 {A[0]->A[1]}");
```

Comment:
The original order of list change moves is affected by this change. Given entity collection `[A, B, C]`, value range `[3, 1, 2]` and solution state:
```
A [2, 1]
B []
C [3]
```
the previous original destination order was:
```
global indices:  0     1     2     3     4     5
                 ⇓     ⇓     ⇓     ⇓     ⇓     ⇓
destinations:   A[0], A[1], A[2], B[0], C[0], C[1]
```
The new approach iterates over entities first and then over values, so the new original order is:
```
entities/values: A     B     C     3     1     2
                 ⇓     ⇓     ⇓     ⇓     ⇓     ⇓
destinations:   A[0], B[0], C[0], C[1], A[2], A[1]

```